cmd/monaco/account/download2.go

```diff
@@ -0,0 +1,146 @@
+
+import (
+	"errors"
+	"fmt"
+	"github.com/dynatrace/dynatrace-configuration-as-code-core/api/clients/accounts"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/cmd/monaco/dynatrace"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/errutils"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/secret"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account/downloader"
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest"
+	manifestloader "github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest/loader"
+	presistance "github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/persistence/account/writer"
+	"github.com/google/uuid"
+	"github.com/spf13/afero"
+	"os"
+)
+
+func downloadAll(fs afero.Fs, opts downloadOpts) error {
+	if opts.outputFolder == "" {
+		opts.outputFolder = "project/accounts" //TODO: make output folder unique (Where to prevent overwriting - write/load module or here?)
+	}
+
+	var accs map[string]manifest.Account
+	if opts.accountUUID == "" {
+		m, errs := manifestloader.Load(&manifestloader.Context{
+			Fs:           fs,
+			ManifestPath: opts.manifestName,
+		})
+		if len(errs) > 0 {
+			errutils.PrintErrors(errs)
+			return errors.New("error while loading manifest")
+		}
+
+		if len(opts.accountName) > 0 {
+			for _, a := range opts.accountName {
+				if n, ok := m.Accounts[a]; !ok {
+					return fmt.Errorf("unknown enviroment %q", n.Name)
+				}
+			}
+			for _, a := range opts.accountName {
+				accs = make(map[string]manifest.Account)
+				accs[a] = m.Accounts[a]
+			}
+		} else {
+			accs = m.Accounts
+		}
+	} else {
+		uuid, err := uuid.Parse(opts.accountUUID)
+		if err != nil {
+			return fmt.Errorf("failed to parese accountUUID: %w", err)
+		}
+		clientID, err := readEnvVariable(opts.clientID)
+		if err != nil {
+			return err
+		}
+		clientSecret, err := readEnvVariable(opts.clientSecret)
+		if err != nil {
+			return err
+		}
+		accs = make(map[string]manifest.Account, 1)
+		accs["account"] = manifest.Account{
+			Name:        "account",
+			AccountUUID: uuid,
+			OAuth: manifest.OAuth{
+				ClientID:     clientID,
+				ClientSecret: clientSecret,
+			},
+		}
+	}
+
+	accountClients, err := dynatrace.CreateAccountClients(accs)
+	if err != nil {
+		return fmt.Errorf("failed to create account clients: %w", err)
+	}
+
+	var failedDownloads []account.AccountInfo
+	for acc, accClient := range accountClients {
+		err := download(fs, opts, acc, accClient)
+		if err != nil {
+			log.Error("Configuration download for account %q failed! Cause: %s", acc, err)
+			failedDownloads = append(failedDownloads, acc)
+		}
+	}
+
+	if len(failedDownloads) > 0 {
+		var es []string
+		for _, t := range failedDownloads {
+			log.Debug("Failed to download account %q (UUID: %q)", t.Name, t.AccountUUID)
```

Comment:
If this is meant to be a summary I'd log them as Error. Collecting things for debug logging seems of limited use to me

Reply:
It is just for UUID (which is only printed in debug-level messages). But we can live without this.